src/core/database/database.service.ts

```diff
@@ -218,48 +226,79 @@ export class DatabaseService {
    */
   async sgUpdateProperties<TObject extends Resource>({
     object,
-    props,
     changes,
     nodevar,
   }: {
     session: Session;
     object: TObject;
-    props: ReadonlyArray<keyof TObject & string>;
     changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> };
     nodevar: string;
   }) {
     return await this.updateProperties({
       type: upperFirst(nodevar),
       object,
-      props,
       changes,
     });
   }
 
+  // expecting the changes of the "simple" properties.
+  async getActualChanges<TResource extends Resource>(
+    oldObject: TResource,
+    changes: ChangesOf<TResource>,
+    props: ReadonlyArray<keyof TResource & string>
+  ): Promise<ChangesOf<TResource>> {
```

Comment:
I'd rather pass in the static resource class here instead of its prop list. This is more consistent with our other functions which accept the same thing. It gives us flexibility in the future to grab more info from the static resource.
It's also terser in use across the app.

```suggestion
  async getActualChanges<TResourceStatic extends ResourceShape>(
    resource: TResourceStatic,
    existingObject: TResourceStatic['prototype'],
    changes: ChangesOf<TResourceStatic['prototype']>,
  ): Promise<ChangesOf<TResourceStatic['prototype']>> {
```

Reply:
@CarsonF I've spent several hours trying to do just that without success, but I'll try again and see.